src/core/lib/transport/metadata_batch.h

```diff
@@ -103,6 +111,27 @@ grpc_error* grpc_metadata_batch_link_tail(grpc_metadata_batch* batch,
 grpc_error* grpc_metadata_batch_add_head(
     grpc_metadata_batch* batch, grpc_linked_mdelem* storage,
     grpc_mdelem elem_to_add) GRPC_MUST_USE_RESULT;
+grpc_error* grpc_metadata_batch_add_head(
+    grpc_metadata_batch* batch, grpc_linked_mdelem* storage,
+    grpc_metadata_batch_callouts_index idx) GRPC_MUST_USE_RESULT;
+grpc_error* grpc_metadata_batch_add_head(
+    grpc_metadata_batch* batch, grpc_linked_mdelem* storage,
+    grpc_mdelem elem_to_add,
+    grpc_metadata_batch_callouts_index idx) GRPC_MUST_USE_RESULT;
+
+inline grpc_error* grpc_metadata_batch_add_head(
```

Comment:
If we have the definition here, why do we need the declaration block above?

Reply:
In this case I was not sure whether GRPC_MUST_USE_RESULT plays well with inline function declarations, so I took this route to be safe.

Reply:
Fixed